controllers/operands/deploymentHandler.go

```diff
@@ -43,27 +43,41 @@ func (h deploymentHooks) getEmptyCr() client.Object {
 
 func (deploymentHooks) justBeforeComplete(_ *common.HcoRequest) { /* no implementation */ }
 
-func (h deploymentHooks) updateCr(req *common.HcoRequest, Client client.Client, exists runtime.Object, _ runtime.Object) (bool, bool, error) {
-	found, ok := exists.(*appsv1.Deployment)
+func (h deploymentHooks) updateCr(req *common.HcoRequest, Client client.Client, exists runtime.Object, required runtime.Object) (bool, bool, error) {
+	deployment, ok1 := required.(*appsv1.Deployment)
+	found, ok2 := exists.(*appsv1.Deployment)
 
-	if !ok {
+	if !ok1 || !ok2 {
 		return false, false, errors.New("can't convert to Deployment")
 	}
-	if !hasCorrectDeploymentFields(found, h.required) {
+	if !hasCorrectDeploymentFields(found, deployment) {
 		if req.HCOTriggered {
 			req.Logger.Info("Updating existing Deployment to new opinionated values", "name", h.required.Name)
 		} else {
 			req.Logger.Info("Reconciling an externally updated Deployment to its opinionated values", "name", h.required.Name)
 		}
-		util.DeepCopyLabels(&h.required.ObjectMeta, &found.ObjectMeta)
-		h.required.DeepCopyInto(found)
-		err := Client.Update(req.Ctx, found)
+		if reflect.DeepEqual(found.Spec.Selector, deployment.Spec.Selector) {
+			// selector hasn't changed, so we can update the resource
```

Comment:
I think we should implement an helper function to detect if we can simply update or if we have to delete and recreate since we are trying to update an immutable field.